common/src/test/java/com/tc/net/core/NoReconnectThreadTest.java

```diff
@@ -117,52 +118,64 @@ public void testConnectionEstablisherThreadExit() throws Exception {
     );
     NetworkListener listener = serverCommsMgr.createListener(new TCSocketAddress(0), true,
                                                              new DefaultConnectionIdFactory(), (MessageTransport t)->true);
-    listener.start(Collections.<ConnectionID>emptySet());
-    int serverPort = listener.getBindPort();
-
-    int proxyPort = new PortChooser().chooseRandomPort();
-    TCPProxy proxy = new TCPProxy(proxyPort, InetAddress.getByName("localhost"), serverPort, 0, false, null);
-    proxy.start();
-
-    ClientMessageChannel client1 = createClientMsgCh();
-    ClientMessageChannel client2 = createClientMsgCh();
-    ClientMessageChannel client3 = createClientMsgCh();
-    
-    InetSocketAddress serverAddress = new InetSocketAddress("localhost", proxyPort);
-
-    client1.addListener(this);
-    client1.open(serverAddress);
-
-    client2.addListener(this);
-    client2.open(serverAddress);
-
-    client3.addListener(this);
-    client3.open(serverAddress);
-
-    ThreadUtil.reallySleep(2000);
-    assertTrue(client1.isConnected());
-    assertTrue(client2.isConnected());
-    assertTrue(client3.isConnected());
-
-    // closing all connections from server side
-    System.err.println("XXX closing all client connections");
-    serverCommsMgr.getConnectionManager().closeAllConnections(1000);
-
-    while (connections.get() != 0) {
-      ThreadUtil.reallySleep(2000);
-      System.err.println(".");
-    }
-
-    // None of the clients should start the ClientConnectionEstablisher Thread for reconnect as the Client
-    // CommsManager is created with reconnect 0. we might need to wait till the created CCE gets quit request.
-    while (getThreadCount(ClientConnectionEstablisher.RECONNECT_THREAD_NAME) > baseAsyncThreads) {
-      ThreadUtil.reallySleep(1000);
-      System.err.println("-");
+    try {
+      listener.start(Collections.emptySet());
```

Comment:
If the 'resource open' is this call then it should be outside the try - no?

Reply:
You are correct; it seems that the call to `start` is what "opens" the listener -- could be moved out of the try block.